docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,2165 @@
+        $container = $this->addProductOfferPostUpdatePlugins($container);
+
+        return $container;
+    }
+
+    /**
+     * @param \Spryker\Zed\Kernel\Container $container
+     *
+     * @return \Spryker\Zed\Kernel\Container
+     */
+    public function provideCommunicationLayerDependencies(Container $container): Container
+    {
+        $container = parent::provideCommunicationLayerDependencies($container);
+
+        return $container;
+    }
+
+    /**
+     * @param \Spryker\Zed\Kernel\Container $container
+     *
+     * @return \Spryker\Zed\Kernel\Container
+     */
+    public function providePersistenceLayerDependencies(Container $container): Container
+    {
+        $container = parent::providePersistenceLayerDependencies($container);
+
+        return $container;
+    }
+
+    /**
+     * @param \Spryker\Zed\Kernel\Container $container
+     *
+     * @return \Spryker\Zed\Kernel\Container
+     */
+    protected function addGlossaryFacade(Container $container): Container
+    {
+        $container->set(static::FACADE_GLOSSARY, function (Container $container) {
+                $container->getLocator()->glossary()->facade()
+        });
+
+        return $container;
+    }
+    
+    /**
+     * @param \Spryker\Zed\Kernel\Container $container
+     *
+     * @return \Spryker\Zed\Kernel\Container
+     */
+    protected function addProductOfferPostUpdatePlugins(Container $container): Container
+    {
+        $container(static::PLUGINS_PRODUCT_OFFER_POST_UPDATE, function (Container $container) {
+            return $this->getProductOfferPostUpdatePlugins($container);
+        });
+
+        return $container;
+    }
+
+    /**
+     * @api
+     *
+     * @param \Spryker\Zed\Kernel\Container $container
+     *
+     * @return array<\Spryker\Zed\ProductOfferExtension\Dependency\Plugin\ProductOfferPostUpdatePluginInterface>
+     */
+    protected function getProductOfferPostUpdatePlugins(Container $container): array
+    {
+        return [];
+    }
+}
+```
+
+## Entity
+
+```
+src
+├── Generated
+│   └── Shared
+│       └── Transfer
+│           └── [EntityName]EntityTransfer.php
+│
+├── Orm
+│   └── Zed   
+│       └── [DomainName]   
+│           ├── Base
+│           │   ├── [EntityName].php
+│           │   └── [EntityName]Query.php 
+│           ├── Map
+│           │   └── [EntityName]TableMap.php
+│           ├── [EntityName].php
+│           └── [EntityName]Query.php
+│
+└── [Organisation]
+    └── Zed
+        └── [ModuleName]
+            └── Persistence
+                └── Propel
+                    ├── Abstract[EntityName].php
+                    └── Abstract[EntityName]Query.php
+```
+
+**Description**
+
+Active record object which represents a row in a table. `Entities` have getter and setter methods to access the underlying data.
+Each `Entity` has a generated identical [Entity Transfer Object](#transfer-object) which can be used during interaction with other layers.
```

Comment:
Something is not accurate here, we should never use Entity Transfer Objects anymore

Reply:
@demkos What is your opinion? I'm not aware of ETO restrictions 🤷

Reply:
Entity transfers were limited to a single module API by this RFC: https://spryker.atlassian.net/wiki/spaces/RFC/pages/184254835/ACCEPTED+RFC+Scope+of+Entity+Transfers